core/src/test/scala/dagr/core/execsystem/TaskManagerTest.scala

```diff
@@ -536,12 +538,15 @@ class TaskManagerTest extends UnitSpec with PrivateMethodTester with OptionValue
     taskManager.addTask(predecessor)
     taskManager.graphNodeStateFor(predecessor).value should be(GraphNodeState.PREDECESSORS_AND_UNEXPANDED)
     taskManager.stepExecution()
+    taskManager.joinOnRunningTasks(1000)
```

Comment:
Consider mixing in a trait to the task manager created by `getDefaultTaskManager` that overrides the `joinOnRunningTasks` and sets it to `1000` or some reasonable default.